src/software/jetson_nano/thunderloop.cpp

```diff
@@ -19,6 +19,31 @@
 extern int clock_nanosleep(clockid_t __clock_id, int __flags,
                            __const struct timespec* __req, struct timespec* __rem);
 
+// signal handling is done by csignal which requires a function pointer with C linkage
+extern "C"
+{
+    static MotorService* g_motor_service = NULL;
+
+    /**
+     * Handles process signals
+     *
+     * @param the signal value (SIGINT, SIGABRT, SIGTERN, etc)
+     */
+    void tbotsExit(int signal_num)
+    {
+        g_motor_service->resetMotorBoard();
+
+        // by now g3log may have died due to the termination signal, so it isn't reliable
+        // to log messages
+        std::cerr << "\n\n!!!\nReceived termination signal: "
```

Comment:
nima: From my understanding, \n doesn't guarantee that the cout buffer is flushed, so it might be safer to use std::endl to make sure it is flushed. A better alternative could be to use std::cerr instead, which logs to the error stream which is not buffered.

Reply:
changed to std::cerr @nimazareian